local/dependencies.go

```diff
@@ -37,29 +37,54 @@ const (
 	ServiceStarted
 )
 
-func inDependencyOrder(ctx context.Context, project *types.Project, fn func(context.Context, types.ServiceConfig) error) error {
+type graphTraversalConfig struct {
+	extremityNodesFn            func(*Graph) []*Vertex                        // leaves or roots
+	adjacentNodesFn             func(*Vertex) []*Vertex                       // getParents or getChildren
+	filterAdjacentByStatusFn    func(*Graph, string, ServiceStatus) []*Vertex // filterChildren or filterParents
+	targetServiceStatus         ServiceStatus
+	adjacentServiceStatusToSkip ServiceStatus
+}
+
+var (
+	upDirectionTraversalConfig = graphTraversalConfig{
+		extremityNodesFn:            leaves,
+		adjacentNodesFn:             getParents,
+		filterAdjacentByStatusFn:    filterChildren,
+		adjacentServiceStatusToSkip: ServiceStopped,
+		targetServiceStatus:         ServiceStarted,
+	}
+	downDirectionTraversalConfig = graphTraversalConfig{
+		extremityNodesFn:            roots,
+		adjacentNodesFn:             getChildren,
+		filterAdjacentByStatusFn:    filterParents,
+		adjacentServiceStatusToSkip: ServiceStarted,
+		targetServiceStatus:         ServiceStopped,
+	}
+)
+
+func inDependencyOrder(ctx context.Context, project *types.Project, traversalConfig graphTraversalConfig, fn func(context.Context, types.ServiceConfig) error) error {
```

Comment:
I wish we had two functions `inDependencyOrder` and `inReverseDependencyOrder`, the `traversalConfig` is confusing

Reply:
Was thinking the same. Policy to define travel within the graph is nice, but this should be hidden with higher level vocabulary.
`inDependecyOrder` vs `inReverseDependencyOrder` would be more readable for the price of an extra nested  func call.